lib/paper_trail/has_paper_trail.rb

```diff
@@ -397,12 +395,12 @@ def record_update(force = nil)
           if pt_record_object_changes?
             data[:object_changes] = pt_recordable_object_changes
           end
-          if self.class.paper_trail_version_class.column_names.include?("transaction_id")
-            data[:transaction_id] = PaperTrail.transaction_id
-          end
+          add_transaction_id(data)
           version = send(self.class.versions_association_name).create merge_metadata(data)
-          update_transaction_id(version)
-          save_associations(version)
+          unless log_version_errors(version, :update)
+            update_transaction_id(version)
+            save_associations(version)
+          end
```

Comment:
That's what I had originally until RuboCop complained about `AbcSize`. I've changed it back, though subsequent edits seem to have removed the need to bump `AbcSize`.